### DISCUSSION TITLE
Deploy with JS dependencies from NPM

### DISCUSSION THREAD

**Discussion by smessmer** [leptos-rs/leptos#1708] [Q&A]
I'd like to include [Milkdown](https://milkdown.dev/) as a component into my leptos web app. Milkdown is a Javascript library available on NPM. I need to run a small JavaScript file initializing that library, something similar to
```js
import { Editor } from '@milkdown/core';

Editor.make()
```

Cargo-leptos offers a way to bundle JavaScript files via `#[wasm_bindgen(module = "/file/path.js")`, so I tried that. But unfortunately it seems wasm-bindgen [doesn't support `import` statements in those files](https://rustwasm.github.io/wasm-bindgen/reference/js-snippets.html#caveats).

What is the best way to get this to work? Ideally, I'm looking for a way for leptos or cargo-leptos to run a bundler for me, run tree-shaking, and then deploy the final artifact. But if that's not possible, I'm also ok with other solutions.

The only way I could think of at the moment is to set up my own webpack config, run webpack from a `build.rs` file, and move the final result somewhere cargo-leptos can find it. But this seems complicated and I feel like "adding npm dependencies" seems like a feature that would be useful to many leptos users. So I was wondering whether leptos already has or is planning to add a better way to do this.

**Comment by ChocolateLoverRaj** [comment]
There is another way of incorporating Webpack by using Webpack instead of Trunk: https://github.com/leptos-rs/leptos/discussions/2333.
